vector/src/main/java/im/vector/app/core/platform/VectorBaseFragment.kt

```diff
@@ -151,6 +151,7 @@ abstract class VectorBaseFragment<VB : ViewBinding> : Fragment(), MavericksView
     override fun onDestroyView() {
         Timber.i("onDestroyView Fragment ${javaClass.simpleName}")
         _binding = null
+        dismissLoadingDialog()
```

Comment:
What happen when the screen is rotated? The dialog is dismissed? But is not displayed back? I am not saying it was the case before...

Reply:
will double check, I imagine it's the same leaking scenario

Reply:
can confirm it's the same behaviour

Forcing a dialog with... 
```kotlin
fun onViewCreated() {
  if (COUNTER == 0) {
    COUNTER++
    showLoadingDialog("Forced loading")
  }
}
```

| BEFORE | AFTER |
| --- | --- |
|![before-dismiss-loading](https://user-images.githubusercontent.com/1848238/146349072-f1a2a23c-1596-467a-84b6-d17f9f885503.gif)|![after-dismiss-dialog](https://user-images.githubusercontent.com/1848238/146349068-e22573d2-c779-437d-8b30-341b6c32f3be.gif)


Before has the leak

```
2021-12-16 09:51:35.951 26636-26636/im.vector.app.debug E/WindowManager: android.view.WindowLeaked: Activity im.vector.app.features.home.HomeActivity has leaked window DecorView@2b4c8fa[HomeActivity] that was originally added here
        at android.view.ViewRootImpl.<init>(ViewRootImpl.java:629)
        at android.view.WindowManagerGlobal.addView(WindowManagerGlobal.java:377)
        at android.view.WindowManagerImpl.addView(WindowManagerImpl.java:96)
        at android.app.Dialog.show(Dialog.java:344)
        at androidx.appcompat.app.AlertDialog$Builder.show(AlertDialog.java:1009)
        at im.vector.lib.ui.styles.dialogs.MaterialProgressDialog.show(MaterialProgressDialog.kt:35)
        at im.vector.lib.ui.styles.dialogs.MaterialProgressDialog.show$default(MaterialProgressDialog.kt:27)
        at im.vector.app.core.platform.VectorBaseFragment.showLoadingDialog(VectorBaseFragment.kt:203)
        at im.vector.app.features.home.HomeDetailFragment.onViewCreated(HomeDetailFragment.kt:131)
        at androidx.fragment.app.Fragment.performViewCreated(Fragment.java:3019)
```

Reply:
So do we want to restore the dialog after a rotation? I would be tempted to say yes, but this is out of scope of this PR which just here to prevent a leak. I merge it, thanks!